### ISSUE TITLE
implement upload file that will be good

### ISSUE THREAD

**Issue by infoBrainSys** [sashabaranov/go-openai#859]
### demo
![image](https://github.com/user-attachments/assets/437a5c84-c3cb-4671-adfb-e9eb7b376df4)

if i can upload my file that will be good, or is there any way can implement this?

**Comment by Carl-Llewellyn** [comment]
Yeah you need to encode the file into whatever GPT takes. E.g. images are encoded in base64. Then you insert it into the prompt.
